eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -393,17 +409,47 @@ object Router {
     def cltvExpiryDelta: CltvExpiryDelta
   }
 
+  // @formatter:off
+  /** source of the channel routing parameters */
+  sealed trait ChannelRelayParams {
+    def cltvExpiryDelta: CltvExpiryDelta
+    def relayFees: Relayer.RelayFees
+    def fee(amount: MilliSatoshi): MilliSatoshi = nodeFee(relayFees, amount)
+    def htlcMinimum: MilliSatoshi
+    def htlcMaximum_opt: Option[MilliSatoshi]
+  }
+
+  object ChannelRelayParams {
+    /** we learnt about this channel from a channel_update */
+    case class FromAnnouncement(channelUpdate: ChannelUpdate) extends ChannelRelayParams {
+      lazy val cltvExpiryDelta: CltvExpiryDelta = channelUpdate.cltvExpiryDelta
+      lazy val relayFees: Relayer.RelayFees = RelayFees(channelUpdate.feeBaseMsat, channelUpdate.feeProportionalMillionths)
+      lazy val htlcMinimum: MilliSatoshi = channelUpdate.htlcMinimumMsat
+      lazy val htlcMaximum_opt: Option[MilliSatoshi] = channelUpdate.htlcMaximumMsat
+    }
+    /** we learnt about this channel from hints in an invoice */
+    case class FromHint(extraHop: Bolt11Invoice.ExtraHop, htlcMaximum: MilliSatoshi) extends ChannelRelayParams {
+      lazy val cltvExpiryDelta: CltvExpiryDelta = extraHop.cltvExpiryDelta
+      lazy val relayFees: Relayer.RelayFees = RelayFees(extraHop.feeBase, extraHop.feeProportionalMillionths)
+      lazy val htlcMinimum: MilliSatoshi = 0 msat
+      lazy val htlcMaximum_opt: Option[MilliSatoshi] = Some(htlcMaximum)
+    }
```

Comment:
Are the `lazy` really useful here?

Reply:
Probably not, replaced by simple `def` in 6fe269081.